projects/eval-adxl355-pmdz/src/main.c

```diff
@@ -0,0 +1,115 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include "no-os/uart.h"
+#include "no-os/gpio.h"
+#include "no-os/spi.h"
+#include "no-os/delay.h"
+#include "no-os/util.h"
+#include "stm32_spi.h"
+#include "stm32_gpio.h"
+#include "stm32_uart.h"
+#include "stm32_uart_stdio.h"
+#include "adxl355.h"
+
+#define SPI_DEVICE_ID	1
+#define SPI_CS			15
+#define SPI_CS_PORT		GPIOA
+
+int main ()
+{
+	struct uart_desc *uart;
+	struct adxl355_dev *adxl355;
+
+	int ret;
+	HAL_Init();
+	SystemClock_Config();
+
+	struct stm32_spi_init_param xsip  = {
+		.chip_select_port = SPI_CS_PORT,
+		.get_input_clock = HAL_RCC_GetPCLK1Freq,
+	};
+
+	struct spi_init_param sip = {
+		.device_id = SPI_DEVICE_ID,
+		.max_speed_hz = 4000000,
+		.bit_order = SPI_BIT_ORDER_MSB_FIRST,
+		.mode = SPI_MODE_0,
+		.extra = &xsip,
+		.platform_ops = &stm32_spi_ops,
+		.chip_select = SPI_CS,
+	};
+
+	struct adxl355_init_param init_data_adxl355 = {
+		.spi_init = sip,
+		.communication_type = ADXL355_SPI_COMM,
+	};
+
+
+	struct stm32_uart_init_param xuip = {
+		.mode = UART_MODE_TX_RX,
+		.timeout = 10,
+	};
+	struct uart_init_param uip = {
+		.device_id = 5,
+		.baud_rate = 115200,
+		.size = UART_CS_8,
+		.parity = UART_PAR_NO,
+		.stop = UART_STOP_1_BIT,
+		.extra = &xuip,
+	};
+
+	ret = uart_init(&uart, &uip);
+	if (ret < 0)
+		goto error;
+
+	stm32_uart_stdio(uart);
+
+	adxl355_init(&adxl355, init_data_adxl355);
+	adxl355_set_odr_lpf(adxl355, ADXL355_ODR_3_906HZ);
+	adxl355_set_op_mode(adxl355, ADXL355_MEASUREMENT);
+
+	float x[32] = {0};
+	float y[32] = {0};
+	float z[32] = {0};
+	float temp = 0;
+	struct adxl355_status_register_flags status_flags = {0};
+	uint8_t fifo_entries = 0;
+	uint8_t data_buffer [288] = {0};
+
+	while(1) {
+
+		adxl355_get_available_fifo_data(adxl355,
+						&fifo_entries,
+						data_buffer,
+						&x[0],
+						&y[0],
+						&z[0]);
+		printf("Number of read entries from the FIFO %d \n", fifo_entries);
```

Comment:
`print_log.h` can be used in the main function.
https://github.com/analogdevicesinc/no-OS/blob/master/include/no-os/print_log.h